src/shogun/base/SGObject.h

```diff
@@ -92,8 +92,6 @@ template <class T> class SGStringList;
 		    AnyParameterProperties(description, param_properties);             \
 		this->m_parameters->add(param, name, description);                     \
 		this->watch_param(name, param, pprop);                                 \
-		if (pprop.get_model_selection())                                       \
-			this->m_model_selection_parameters->add(param, name, description); \
 		if (pprop.get_gradient())                                              \
 			this->m_gradient_parameters->add(param, name, description);        \
 	}
```

Comment:
Did you try to remove `m_gradient_parameters` yet? Just curious as to how many things break..

Reply:
No, I didn't remove m_gradient_parameters. Mostly the model selection functionality breaks.